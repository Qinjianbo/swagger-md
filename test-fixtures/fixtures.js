import { create as createFixtureLoader } from 'fixture-loader';

const fixture_loader = createFixtureLoader('./test-fixtures/fixtures');

const api = {

  loadSwaggerSpec(filename) {
    return fixture_loader.loadParsedJson('swagger', filename);
  },

};

export default api;
