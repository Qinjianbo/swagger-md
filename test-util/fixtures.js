import { create as createFixtureLoader } from 'fixture-loader';

const fixture_loader = createFixtureLoader('./test-util/fixtures');

const api = {

  loadSwaggerSpec(filename) {
    return fixture_loader.loadParsedJson('swagger', filename);
  },

};

export default api;
