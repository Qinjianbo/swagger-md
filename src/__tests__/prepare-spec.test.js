import prepareSpec from '../prepare-spec';
import fixtures from '../../test-fixtures/fixtures';

describe('prepareSpec', () => {

  describe('with a valid Swagger spec', () => {

    it('should inline all $ref references', () => {
      const swagger_api_spec = fixtures.loadSwaggerSpec('pet-store');
      return prepareSpec(swagger_api_spec).then(spec => {
        expect(spec).toMatchSnapshot();
      });
    });

  });

  describe('with a $ref that cannot be resolved', () => {

    it('should reject with an error', () => {
      const promise = prepareSpec({ $ref: '#/Pet' });
      const must_reject = flipPromise(promise);
      return must_reject.then(err => {
        expect(err.message).toMatch('Error resolving $ref pointer');
        expect(err.message).toMatch('Token "Pet" does not exist');
      });
    });

  });

});


function flipPromise(p) {
  return p.then(() => Promise.reject(new Error('Expected promise to be rejected')), err => err);
}
