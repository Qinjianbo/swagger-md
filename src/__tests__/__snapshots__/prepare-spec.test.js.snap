exports[`prepareSpec with a valid Swagger spec should inline all $ref references 1`] = `
Object {
  "basePath": "/api",
  "consumes": Array [
    "application/json",
  ],
  "definitions": Object {
    "Error": Object {
      "properties": Object {
        "code": Object {
          "format": "int32",
          "type": "integer",
        },
        "message": Object {
          "type": "string",
        },
      },
      "required": Array [
        "code",
        "message",
      ],
      "type": "object",
    },
    "NewPet": Object {
      "description": "A pet",
      "properties": Object {
        "name": Object {
          "description": "The pet\'s name",
          "type": "string",
        },
        "tag": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
    "Pet": Object {
      "allOf": Array [
        Object {
          "description": "A pet",
          "properties": Object {
            "name": Object {
              "description": "The pet\'s name",
              "type": "string",
            },
            "tag": Object {
              "type": "string",
            },
          },
          "required": Array [
            "name",
          ],
          "type": "object",
        },
        Object {
          "properties": Object {
            "id": Object {
              "format": "int64",
              "type": "integer",
            },
          },
          "required": Array [
            "id",
          ],
          "type": "object",
        },
      ],
      "type": "object",
    },
  },
  "host": "petstore.swagger.io",
  "info": Object {
    "contact": Object {
      "email": "foo@example.com",
      "name": "Swagger API Team",
      "url": "http://madskristensen.net",
    },
    "description": "A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification",
    "license": Object {
      "name": "MIT",
      "url": "http://github.com/gruntjs/grunt/blob/master/LICENSE-MIT",
    },
    "termsOfService": "http://swagger.io/terms/",
    "title": "Swagger Petstore",
    "version": "1.0.0",
  },
  "paths": Object {
    "/pets": Object {
      "get": Object {
        "description": "Returns all pets from the system that the user has access to",
        "operationId": "findPets",
        "parameters": Array [
          Object {
            "collectionFormat": "csv",
            "description": "tags to filter by",
            "in": "query",
            "items": Object {
              "enum": Array [
                "clueless",
                "lazy",
                "adventurous",
                "aggressive",
              ],
              "type": "string",
            },
            "maxItems": 3,
            "minItems": 0,
            "name": "tags",
            "required": false,
            "type": "array",
            "uniqueItems": true,
          },
          Object {
            "default": 20,
            "description": "maximum number of results to return",
            "exclusiveMinimum": true,
            "format": "int32",
            "in": "query",
            "maximum": 200,
            "minimum": 0,
            "name": "limit",
            "required": false,
            "type": "integer",
          },
          Object {
            "description": "results offset",
            "format": "int32",
            "in": "query",
            "minimum": 0,
            "name": "offset",
            "required": false,
            "type": "integer",
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "pet response",
            "schema": Object {
              "items": Object {
                "allOf": Array [
                  Object {
                    "description": "A pet",
                    "properties": Object {
                      "name": Object {
                        "description": "The pet\'s name",
                        "type": "string",
                      },
                      "tag": Object {
                        "type": "string",
                      },
                    },
                    "required": Array [
                      "name",
                    ],
                    "type": "object",
                  },
                  Object {
                    "properties": Object {
                      "id": Object {
                        "format": "int64",
                        "type": "integer",
                      },
                    },
                    "required": Array [
                      "id",
                    ],
                    "type": "object",
                  },
                ],
                "type": "object",
              },
              "type": "array",
            },
          },
          "default": Object {
            "description": "unexpected error",
            "schema": Object {
              "properties": Object {
                "code": Object {
                  "format": "int32",
                  "type": "integer",
                },
                "message": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "code",
                "message",
              ],
              "type": "object",
            },
          },
        },
      },
      "post": Object {
        "description": "Creates a new pet in the store.  Duplicates are allowed",
        "operationId": "addPet",
        "parameters": Array [
          Object {
            "description": "Pet to add to the store",
            "in": "body",
            "name": "pet",
            "required": true,
            "schema": Object {
              "description": "A pet",
              "properties": Object {
                "name": Object {
                  "description": "The pet\'s name",
                  "type": "string",
                },
                "tag": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "name",
              ],
              "type": "object",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "pet response",
            "schema": Object {
              "allOf": Array [
                Object {
                  "description": "A pet",
                  "properties": Object {
                    "name": Object {
                      "description": "The pet\'s name",
                      "type": "string",
                    },
                    "tag": Object {
                      "type": "string",
                    },
                  },
                  "required": Array [
                    "name",
                  ],
                  "type": "object",
                },
                Object {
                  "properties": Object {
                    "id": Object {
                      "format": "int64",
                      "type": "integer",
                    },
                  },
                  "required": Array [
                    "id",
                  ],
                  "type": "object",
                },
              ],
              "type": "object",
            },
          },
          "default": Object {
            "description": "unexpected error",
            "schema": Object {
              "properties": Object {
                "code": Object {
                  "format": "int32",
                  "type": "integer",
                },
                "message": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "code",
                "message",
              ],
              "type": "object",
            },
          },
        },
      },
    },
    "/pets/{id}": Object {
      "delete": Object {
        "deprecated": true,
        "description": "deletes a single pet based on the ID supplied",
        "operationId": "deletePet",
        "parameters": Array [
          Object {
            "description": "ID of pet to delete",
            "format": "int64",
            "in": "path",
            "name": "id",
            "required": true,
            "type": "integer",
          },
        ],
        "responses": Object {
          "204": Object {
            "description": "pet deleted",
          },
          "default": Object {
            "description": "unexpected error",
            "schema": Object {
              "properties": Object {
                "code": Object {
                  "format": "int32",
                  "type": "integer",
                },
                "message": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "code",
                "message",
              ],
              "type": "object",
            },
          },
        },
      },
      "get": Object {
        "description": "Returns a user based on a single ID, if the user does not have access to the pet",
        "operationId": "find pet by id",
        "parameters": Array [
          Object {
            "description": "ID of pet to fetch",
            "format": "int64",
            "in": "path",
            "name": "id",
            "required": true,
            "type": "integer",
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "pet response",
            "schema": Object {
              "allOf": Array [
                Object {
                  "description": "A pet",
                  "properties": Object {
                    "name": Object {
                      "description": "The pet\'s name",
                      "type": "string",
                    },
                    "tag": Object {
                      "type": "string",
                    },
                  },
                  "required": Array [
                    "name",
                  ],
                  "type": "object",
                },
                Object {
                  "properties": Object {
                    "id": Object {
                      "format": "int64",
                      "type": "integer",
                    },
                  },
                  "required": Array [
                    "id",
                  ],
                  "type": "object",
                },
              ],
              "type": "object",
            },
          },
          "default": Object {
            "description": "unexpected error",
            "schema": Object {
              "properties": Object {
                "code": Object {
                  "format": "int32",
                  "type": "integer",
                },
                "message": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "code",
                "message",
              ],
              "type": "object",
            },
          },
        },
      },
    },
  },
  "produces": Array [
    "application/json",
  ],
  "schemes": Array [
    "http",
  ],
  "swagger": "2.0",
}
`;
